libc/string_chr.c

```diff
@@ -0,0 +1,253 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ * 		- string.h
+ * TESTED:
+ *    - strchr()
+ *    - strrchr()
+ *    - memchr()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Damian Modzelewski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <string.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <limits.h>
+#include <unity_fixture.h>
+
+#define LOREM_IPSUM "Lorem ipsum dolor sit amet, consectetur adipiscing elit"
+
+TEST_GROUP(string_strchr);
+
+
+TEST_SETUP(string_strchr)
+{
+}
+
+
+TEST_TEAR_DOWN(string_strchr)
+{
+}
+
+
+TEST(string_strchr, strchr_standard_use)
+{
+	char str[PATH_MAX] = "a";
+	/*Check of proper working base*/
+	TEST_ASSERT_EQUAL_STRING(LOREM_IPSUM, strchr(LOREM_IPSUM, 'L'));
+	TEST_ASSERT_EQUAL_STRING("lor sit amet, consectetur adipiscing elit", strchr(LOREM_IPSUM, 'l'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(LOREM_IPSUM, 'x'));
+
+	for (int i = 0; i < PATH_MAX - 5; i++) {
+
+		if (i == PATH_MAX - 10) {
+			strcat(str, "b");
+		}
+		else
+			strcat(str, "a");
+	}
```

Comment:
```suggestion
	memset(str, 'a', PATH_MAX - 5);
	str[PATH_MAX - 10] = 'b';
```

Much shorter. Though it needs `str` declaration as follows:
```c
char str[PATH_MAX] = { 0 };
```

Also:
 - `-5` and `-10` seems rather arbitrary. 
 - this test may be split into two: one with _loremipsum_ usage and other with `PATH_MAX` case. But that`s just a suggestion.
 - it is unclear why we use one type of string for forward search (`strchr`) but very different string for reverse search (strrchr).

Reply:
Or if you`re very eager on not using `memset` you can just:
```suggestion
	for (i = 0; i < PATH_MAX - 5; i++) {
		str[i] = 'a';
	}
	str[PATH_MAX - 10] = 'b';
```